src/Cms/ModelWithContent.php

```diff
@@ -524,16 +528,21 @@ public function query(string $query = null, string $expect = null)
 	 */
 	public function readContent(string $languageCode = null): array
 	{
-		$file = $this->contentFile($languageCode);
+		$identifier = $this::CLASS_ALIAS === 'page' && $this->isDraft() === true ?
+			VersionIdentifier::changes() :
+			VersionIdentifier::published();
 
-		// only if the content file really does not exist, it's ok
-		// to return empty content. Otherwise this could lead to
-		// content loss in case of file reading issues
-		if (file_exists($file) === false) {
+		try {
+			return $this->storage()->read(
+				$identifier,
+				$this->storage()->languageCodeToObject($languageCode)
+			);
+		} catch (NotFoundException $e) {
+			// only if the content file really does not exist, it's ok
+			// to return empty content. Otherwise this could lead to
+			// content loss in case of file reading issues
 			return [];
```

Comment:
TBH I disagree. From my comment above:

> I'm not sure if it's a good idea to silently "not do anything" in an error case. Dropping too many exceptions or silently correcting method input has bitten us quite often in the past.

Here in this case we return an empty array for backwards-compatibility, but that may not be wanted in every case. Or actually it may not be wanted at all, we just have to do it here to avoid breakage.

Keep in mind that `readContent()` is basically deprecated because it doesn't support the changes version, but we can't formally add a deprecation comment or warning yet because there is no user-facing replacement until the content storage layer is stable and no longer internal.